frontend/src/pages/sites/sites.tsx

```diff
@@ -36,45 +39,63 @@ const Sites: React.FC = () => {
   useTitle('My sites');
 
   const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [search, setSearch] = useState<string>('');
+
+  const { sites, status, sitesCount } = useAppSelector(({ sites }) => ({
+    sites: sites.sites,
+    status: sites.dataStatus,
+    sitesCount: sites.sitesCount,
+  }));
+
+  const { page, handleChangePage, totalPages, isShowPagination } =
+    usePagination({
+      pageDefault: DEFAULT_PAGE,
+      rowsPerPage: SITES_PER_PAGE,
+      totalCount: sitesCount,
+    });
 
   useEffect((): void => {
     if (projectId) {
       void dispatch(
         sitesActions.getSitesByProjectId({
           parameters: { projectId: Number(projectId) },
           queryParameters: {
-            name: '',
+            name: search,
```

Comment:
by initial load it can be empty. so you don't need a state for it

Reply:
This effect handles pagination too, so I need save search string somehow

Reply:
getValues from react-hook-form also a solution

Reply:
prefer `getValues `